Makefile

```diff
@@ -0,0 +1,18 @@
+# Docker is the only dependency here
+# Build the dashboard image for host or cross-compile
+# This Makefile can only be run on a amd64 host
+
+build-image:
```

Comment:
This introduces a new build system to the project. Currently we have Gulp + a few bash scripts. With this change there's also make. This would result in confusion for developers on what to use and there'll be additional maintenance burden to maintain additional build system. 

My thinking is that you come from a different background and you basically like/know Makefiles better. That's perfectly okay. Using Gulp, however, is not that hard when you know JS.

I think that all you care about is that you dont want to install all the dependencies and want to cross compile the code for many platforms. These are very good use cases. 

How about making the API for cross build like this. You git clone the repo and `./build/run-gulp-in-docker.sh [ARGUMENT_FOR_GULP]`, e.g., `./build/run-gulp-in-docker.sh cross-compile`. The bash script would basically run our gulp pipeline inside a docker image that has all dependencies installed. There will be a `cross-compile` task in gulp that produces multiple images (this will have a huge speedup, btw, as only backend would have to be compiled a few times, frontend not).

What do you think?

CC @floreks @cheld @maciaszczykm